api/worker/handlers.go

```diff
@@ -30,7 +30,7 @@ func (a *WorkerApi) AddWorker(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if err = a.auth.Validate([]byte(signature), worker); err != nil {
+	if err = a.auth.Authenticate([]byte(signature), worker); err != nil {
```

Comment:
wouldn't we return a partially formed token from Authenticate?

Reply:
Done.